Source/DafnyCore/ProofDependencyWarnings.cs

```diff
@@ -95,6 +176,98 @@ private static void Warn(DafnyOptions dafnyOptions, ErrorReporter reporter, Proo
         }
       }
     }
+
+    if (dafnyOptions.Get(CommonOptionBag.SuggestProofRefactoring) && depManager.idsByMemberName[scopeName].Decl is Method m) {
+      // Function hiding suggestion
+      if (unusedFunctions.Any()) {
+        reporter.Info(MessageSource.Verifier, m.Body.StartToken,
+          $"Consider hiding {(unusedFunctions.Count > 1 ? "these functions, which are" : "this function, which is")} unused by the proof: {unusedFunctions.Comma()}");
+      }
+
+      SuggestByProofRefactoring(reporter, depManager, scopeName, assertCoverage);
+    }
+  }
+
+  private static void SuggestByProofRefactoring(ErrorReporter reporter, ProofDependencyManager depManager,
```

Comment:
Instead of `List<(IEnumerable<TrackedNodeComponent> CoveredElements, HashSet<(Boogie.IToken Tok, string Description, string Id)> Asserts)>`, can it be `IReadOnlyList<VCResultLogEntry>` ?

Reply:
`WarnAboutSuspiciousDependenciesForImplementation` only has `IReadOnlyList<VerificationRunResult>`.

Reply:
Sure but `results.Select(DafnyConsolePrinter.DistillVCResult).ToList()` should turn it into a `IReadOnlyList<VerificationRunResult>`, wouldn't it?